### DISCUSSION TITLE
Do initonly properties work for factory methods?

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/csharplang#4034] [Q&A] [answered]
Assume I have a struct

``` CS
public struct Point
{
  public double X  {get; init;}
  public double Y  {get; init;}
}

```

I assume the init-only does not work for factory methods:

``` CS
public static Point Create() => new Point();
``` 

so I cannot write

```CS
  Create().X = 10;
```

correct? 
If it doesn't work would it be worth a proposal such that factory methods still allow access to init only properties? I know this bears a lot of problems, but I think it's worth to think about it.

**Comment by Joe4evr** [comment, marked as answer]
I don't think it's going to be in 9.0, but the idea that the LDT wants to do is allowing to annotate the factory method with an attribute (eg. `[Factory]`) which the compiler recognizes and will limit what you can actually return (only `new` and the return value of other `[Factory]` methods), and that will let you add an object initializer at the callsite. [More information here](https://youtu.be/fIXujTOA6BE?t=3023) (the example is on `With()`, but that's just a special case of the more general feature).

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-20.md

> We'll consider broader Factory scenarios later.

**Comment by msedi** [comment]
Thanks to both of you.
